src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1006,30 +1006,14 @@ bool SQLStorage::loadCachedEcuManifest(const Uptane::EcuSerial& ecu_serial, std:
   return !empty;
 }
 
-void SQLStorage::storeMisconfiguredEcus(const std::vector<MisconfiguredEcu>& ecus) {
-  if (ecus.size() >= 1) {
-    SQLite3Guard db = dbConnection();
-
-    db.beginTransaction();
-
-    if (db.exec("DELETE FROM misconfigured_ecus;", nullptr, nullptr) != SQLITE_OK) {
-      LOG_ERROR << "Can't clear misconfigured_ecus: " << db.errmsg();
-      return;
-    }
-
-    std::vector<MisconfiguredEcu>::const_iterator it;
-    for (it = ecus.begin(); it != ecus.end(); it++) {
-      auto statement = db.prepareStatement<std::string, std::string, int>(
-          "INSERT INTO misconfigured_ecus VALUES (?,?,?);", it->serial.ToString(), it->hardware_id.ToString(),
-          static_cast<int>(it->state));
-
-      if (statement.step() != SQLITE_DONE) {
-        LOG_ERROR << "Can't set misconfigured_ecus: " << db.errmsg();
-        return;
-      }
-    }
+void SQLStorage::saveMisconfiguredEcu(const MisconfiguredEcu& ecu) {
+  SQLite3Guard db = dbConnection();
 
-    db.commitTransaction();
+  auto statement = db.prepareStatement<std::string, std::string, int>(
+      "INSERT OR REPLACE INTO misconfigured_ecus VALUES (?,?,?);", ecu.serial.ToString(), ecu.hardware_id.ToString(),
+      static_cast<int>(ecu.state));
+  if (statement.step() != SQLITE_DONE || sqlite3_changes(db.get()) != 1) {
```

Comment:
I'm not sure it's worth checking `sqlite3_changes` value here. My understanding was that it is only useful when you have `WHERE` in your statement or something similar.

Reply:
Oops, that was lazy copying on my part, didn't even think about it. I'll double check but will probably remove it.

Reply:
Fixed. Not is it not necessary, but according to https://www.sqlite.org/c3ref/changes.html it doesn't even do anything for `REPLACE`, so it would've caused a bug.

Reply:
That was my first impression as well, but it looks like while `REPLACE` doesn't increment the counter, the subsequent `INSERT` does. As I understood, `REPLACE` does only the deleting part.

Reply:
Oh, I see. Well, I removed it anyway. :) Seems unnecessary.